methods/boin/boinsim.R

```diff
@@ -0,0 +1,144 @@
+# required packages
+
+library(magrittr)
+library(escalation)
+library(ggplot2)
+
+# lists
+
+n_sims <- list()
+
+model <- list()
+sims <- list()
+
+selection <- list()
+selection_df <- list()
+
+treated <- list()
+treatedpct <- list()
+treatedpct_df <- list()
+
+dist_accuracy <- list()
+mean_accuracy <- list()
+
+dist_overdose <- list()
+mean_overdose <- list()
+
+dist_length <- list()
+mean_length <- list()
+
+# variables needed (UI)
+
+n_doses <- 5
+ttl <- 1/3
+max_n <- 24
+start_dose <- 1
+current_seed <- 12345
+n_sims$boin <- 100
+true_dlt_ss <- c(0.05, 0.15, 1/3, 0.5, 0.8)
+cohort_size <- 3
+stop_n_mtd <- 15
+
+## NOT YET COLLECTED IN Q_DATABASE
+p_tox <- 0.6
+p_saf <- 0.1
+use_stopping_rule <- TRUE
+n_cohorts <- 10
+
+best_dose <- max(true_dlt_ss[true_dlt_ss<=ttl])
+best_dose_level <- match(best_dose,true_dlt_ss)
+
+# model
+model$boin <- escalation::get_boin(
+  num_doses = n_doses,
+  target = ttl,
+  use_stopping_rule = use_stopping_rule,
+  ncohort = n_cohorts,
+  cohortsize = cohort_size,
+  n.earlystop = stop_n_mtd,
+  p.saf = p_saf, p.tox = p_tox
+)
+
+model$boin <- escalation::get_boin(
+  num_doses = n_doses,
+  target = ttl,
+  use_stopping_rule = use_stopping_rule,
+  p.saf = p_saf, p.tox = p_tox
+  ) %>%
+  stop_when_n_at_dose(n=stop_n_mtd, dose = "recommended") %>%
+  stop_at_n(n = max_n)
+
+# run sims
+
+sims$boin <- model$boin %>%
+  escalation::simulate_trials(next_dose = start_dose, num_sims = n_sims$boin, true_prob_tox = true_dlt_ss, true_prob_eff = NULL)
+
```

Comment:
This part should be consistent with Jake's function for model creation and simulation processing